gtsam_extra.cmake.in

```diff
@@ -1,11 +1,5 @@
 # Extra CMake definitions for GTSAM
 
-set (GTSAM_VERSION_MAJOR @GTSAM_VERSION_MAJOR@)
-set (GTSAM_VERSION_MINOR @GTSAM_VERSION_MINOR@)
-set (GTSAM_VERSION_PATCH @GTSAM_VERSION_PATCH@)
-set (GTSAM_VERSION_NUMERIC @GTSAM_VERSION_NUMERIC@)
-set (GTSAM_VERSION_STRING "@GTSAM_VERSION_STRING@")
-
```

Comment:
Any reason these were taken out? It may break compatibility with existing build systems that use GTSAM, and honestly, having these variables available makes it easy for fine-grained configurations.

Reply:
I removed them because according to the extract below found on [find_package's documentation](https://cmake.org/cmake/help/v3.14/command/find_package.html) they will be set by CMake:

```
 If the version is acceptable the following variables are set:
 
 <PackageName>_VERSION
     full provided version string
 <PackageName>_VERSION_MAJOR
     major version if provided, else 0
 <PackageName>_VERSION_MINOR
     minor version if provided, else 0
 <PackageName>_VERSION_PATCH
     patch version if provided, else 0
 <PackageName>_VERSION_TWEAK
     tweak version if provided, else 0
 <PackageName>_VERSION_COUNT
     number of version components, 0 to 4
 
 and the corresponding package configuration file is loaded.
```

Currently `GTSAMConfig.cmake` is set to load `gtsam_extra.cmake` that would define these variables too, so I thought it would be better to remove them from this file and let CMake set them for us.

About compatibility, I don't think this variables are passed downstream. I looked at the code and I only saw them being used in configuration files, in which they are replaced by their value during configuration. The `FindGTSAM.cmake` script provided in the repo doesn't seem to use it either.

By the way, `FindGTSAM.cmake` doesn't seem to handle versioning neither, as I see no reference to these `GTSAM_VERSION_*` variables there nor a `VERSION_VAR` being passed to `find_package_handle_standard_args` (see [doc](https://cmake.org/cmake/help/v3.14/module/FindPackageHandleStandardArgs.html)). I think it should parse the version from `include/gtsam/config.h` and feed it to `find_package_handle_standard_args`. I may do something about it if I have time.

About fine-grained configurations, these variables would still be available downstream, so I don't think it would be a problem.

PS: While writting this I noted that CMake sets `<PackageName>_VERSION` and not `<PackageName>_VERSION_STRING` as it is done in GTSAM. This affects one part of the PR. I'll update it.

Reply:
Fixed `GtsamMakeConfigFile.cmake` to use `PACKAGE_NAME` argument instead of `GTSAM` and to handle packages with `${PACKAGE_NAME}_VERSION` or `${PACKAGE_NAME}_VERSION_STRING` variables.

Also added handling of versions to `FindGTSAM*.cmake` scripts. I saw they were obsolete, but we never now if someone is using it... ¯\_(ツ)_/¯

Reply:
If really needed I think we could find a way to recover `GTSAM_VERSION_NUMERIC` and `GTSAM_VERSION_STRING` from `gtsam/config.h` to pass downstream.

Reply:
Both `NUMERIC` and `STRING` version variables are now recovered from `gtsam/config.h`

Reply:
I just realized that recovering the variables from config.h is equivalent to setting it directly... :sweat_smile: Anyway, I understand your point on keeping them, just need to decide the way to set them.

Reply:
> Currently GTSAMConfig.cmake is set to load gtsam_extra.cmake that would define these variables too, so I thought it would be better to remove them from this file and let CMake set them for us.

Really nice job investigating and doing the necessary leg work!